src/testSession.ts

```diff
@@ -270,13 +281,18 @@ export class TestSession {
                 this.orgs.push(username);
               }
             }
-            this.setup.push(jsonOutput);
+            this.setup.push({
+              command: cmd,
+              jsonOutput: ensureAnyJson(jsonOutput),
+              shellOutput: rv,
+              execCmdDuration,
+            });
```

Comment:
I'm not really opposed to this but don't feel that it provides much benefit.  How are you using it?

Reply:
From setup in org test
```
    session.setup?.forEach((result) => {
      if (result.command.includes('--setdefaultusername')) {
        defaultUsername = getString(result, 'jsonOutput.result.username');
      }
      if (result.command.includes('--nonamespace')) {
        nonamespaceUsername = getString(result, 'jsonOutput.result.username');
      }
    });
```
The original returned either AyyJson[] | ShellString, what is always available is ShellString and sometimes json. I see it useful to have both, if available. Provides entire view of the command's execution state.

Reply:
The test author knows the array order of the setup commands so you could just assign based on that.
e.g., `const defaultUsername = getString(session, 'setup[1].jsonOutput.result.username');`
All setup commands must return with a zero exit code or all of setup fails and tests aren't run so we can rely on it.

Alternatively,
1. use an alias
2. specify the username when creating the org